test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,303 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import subprocess
+import unittest
+from dataclasses import dataclass
+from textwrap import dedent
+
+import pytest
+
+from materialize.cloudtest.app.materialize_application import MaterializeApplication
+
+
+@dataclass
+class ClusterSetup:
+    storage_replication_factor: int = 1
+    compute_replication_factor: int = 1
+    storage_name = "c_storage"
+    compute_name = "c_compute"
+    # use different availability zones for the two clusters to avoid that they end up on the same node
+    storage_az = "1"
+    compute_az = "2"
+
+
+REQUIRED_PROGRESS = 5
+
+
+def populate(mz: MaterializeApplication, cluster_setup: ClusterSetup) -> None:
+    mz.environmentd.sql(
+        dedent(
+            f"""
+            ALTER SYSTEM SET enable_managed_clusters = true;
+
+            DROP MATERIALIZED VIEW IF EXISTS mv;
+            DROP SOURCE IF EXISTS source;
+            DROP CLUSTER IF EXISTS {cluster_setup.storage_name};
+            DROP CLUSTER IF EXISTS {cluster_setup.compute_name};
+
+            CREATE CLUSTER {cluster_setup.storage_name} SIZE = '1', REPLICATION FACTOR {cluster_setup.storage_replication_factor}, AVAILABILITY ZONES ['{cluster_setup.storage_az}'];
```

Comment:
This unfortunately does not seem to be correct. If you have replicationfactor = 2, one availability zone is in the list. This causes both replicas to be placed in the same availability zone.